presentation/common/src/presentation-common/KeySet.ts

```diff
@@ -165,9 +168,10 @@ export class KeySet {
       if (!set) {
         set = new Set();
         this._instanceKeys.set(entry["0"], set);
+        this._lowerCaseMap.set(entry["0"], (keyset as KeySet)._lowerCaseMap.get(entry["0"])!);
       }
       entry["1"].forEach((id: Id64String) => {
-        if (!pred || pred({ className: entry[0], id }))
+        if (!pred || pred({ className: (keyset as KeySet)._lowerCaseMap.get(entry["0"])!, id }))
```

Comment:
It's very inefficient to lookup with the class name on every id. The lookup could be made once at the top of the method.

Reply:
Yeeh, fixed it